test/utils/objects/objects.go

```diff
@@ -113,6 +115,17 @@ func TestKC(matchLabels map[string]string) (*machineconfigv1.KubeletConfig, erro
 	}, nil
 }
 
+func IsOwnedBy(obj metav1.ObjectMeta, owner metav1.ObjectMeta) bool {
```

Comment:
this works and I'm fine with it. I'd have used the daemonset names reported in the Status, though. No need to change, I don't have strong opinions and this approach has no obvious issues.